### ISSUE TITLE
[BUG] tracer: unable to trace if FQDN is used as an advertised address for a socket

### ISSUE THREAD

**Issue by ovidiusas** [OpenSIPS/opensips#3340]
**OpenSIPS version you are running**
Latest master and 3.4.

**Describe the bug**
If an FQDN is used as an advertised address for a socket, tracing will fail inside pipport2su() from tracer.c with error:
```
LM_ERR("host <%.*s> is not an IP\n",host_uri.len,host_uri.s);
```

A simple way to reproduce this:
```
socket=[my_proto]:[my_ip] as [my_fqdn]
```
Then perform tracing inside the config.
This is a side effect of commit a13e034.

@vladpaiu, can you please take a look at this issue?
When we are using the `adv_address` for tracing we should check if it's an IP or not (using `str2ip` and `str2ip6`) and only after that to use it for tracing. If it's not an IP, then we should either default to non advertised IP or resolve the FQDN and use the resolved IP for tracing.

**Comment by ovidiusas** [comment]
Or maybe we should try to resolve the FQDN inside pipport2su() before throwing the "is not IP" error.

**Comment by ovidiusas** [comment]
Please check pull request #3352
It has a fix for this issue.

**Comment by bogdan-iancu** [comment]
@ovidiusas , could you generate a new PR with the proposed fix here ?

**Comment by bogdan-iancu** [comment]
@ovidiusas , I guess the fix you push is a complete fix for the reported issue here, right ? Just to know if we should close here too

**Comment by ovidiusas** [comment]
All is good now. The backport is also good!
